packages/instrumentation-sequelize/src/sequelize.ts

```diff
@@ -0,0 +1,137 @@
+} from '@opentelemetry/instrumentation';
+
+type Config = InstrumentationConfig & SequelizeInstrumentationConfig;
+
+export class SequelizeInstrumentation extends InstrumentationBase<typeof sequelize> {
+    static readonly component = 'sequelize';
+    protected _config!: Config;
+
+    constructor(config: Config = {}) {
+        super('opentelemetry-instrumentation-sequelize', VERSION, Object.assign({}, config));
+    }
+
+    setConfig(config: Config = {}) {
+        this._config = Object.assign({}, config);
+        if (config.logger) this._logger = config.logger;
+    }
+
+    protected init(): InstrumentationModuleDefinition<typeof sequelize> {
+        const module = new InstrumentationNodeModuleDefinition<typeof sequelize>(
+            SequelizeInstrumentation.component,
+            ['*'],
+            this.patch.bind(this),
+            this.unpatch.bind(this)
+        );
+        return module;
+    }
+
+    protected patch(moduleExports: typeof sequelize): typeof sequelize {
+        if (moduleExports === undefined || moduleExports === null) {
+            return moduleExports;
+        }
+
+        this._logger.debug(`applying patch to sequelize`);
+        this.unpatch(moduleExports);
+        this._wrap(moduleExports.Sequelize.prototype, 'query', this._createQueryPatch.bind(this));
+
+        return moduleExports;
+    }
+
+    protected unpatch(moduleExports: typeof sequelize): void {
+        if (isWrapped(moduleExports.Sequelize.prototype.query)) {
+            this._unwrap(moduleExports.Sequelize.prototype, 'query');
+        }
+    }
+
+    private _createQueryPatch(original: Function) {
+        const thisInstrumentation = this;
+        return function (sql: any, option: any) {
+            if (thisInstrumentation._config?.ignoreOrphanedSpans && !getSpan(context.active())) {
+                return original.apply(this, arguments);
+            }
+
+            let statement = sql?.query ? sql.query : sql;
+            let operation = option.type;
+
+            if (typeof statement === 'string') {
+                statement = statement.trim();
+                if (!operation) operation = statement.split(' ')[0];
+            }
+
+            const sequelizeInstance: sequelize.Sequelize = this;
+            const config = sequelizeInstance?.config;
+
+            const attributes = {
+                [DatabaseAttribute.DB_SYSTEM]: sequelizeInstance.getDialect(),
+                [DatabaseAttribute.DB_USER]: config?.username,
+                [GeneralAttribute.NET_PEER_NAME]: config?.host,
+                [GeneralAttribute.NET_PEER_PORT]: config?.port ? Number(config?.port) : undefined,
+                [GeneralAttribute.NET_TRANSPORT]: thisInstrumentation._getNetTransport(config?.protocol),
+                [DatabaseAttribute.DB_NAME]: config?.database,
+                [DatabaseAttribute.DB_OPERATION]: operation,
+                [DatabaseAttribute.DB_STATEMENT]: statement,
+                component: 'sequelize',
```

Comment:
good time to remove this attribute, before it becomes a breaking change

Reply:
in a different PR